skimage/feature/_kadir_brady.py

```diff
@@ -0,0 +1,325 @@
+        cluster[2, 0] = t_scale[index]
+        s_i = np.argsort(D[index, :])
+        # fill in the neighbouring regions
+        for j in range(k):
+            cluster[0, j+1] = t_c[s_i[j+1]]
+            cluster[1, j+1] = t_r[s_i[j+1]]
+            cluster[2, j+1] = t_scale[s_i[j+1]]
+
+        # clusters center point
+        center = np.array([np.mean(cluster, axis=1)])
+
+        # check if the regions are "suffiently clustered", if variance is less than threshold
+        v = np.var(np.sqrt(((cluster - np.tile(center.T, (1, k + 1))) ** 2).sum(axis=0)))
+        if v > v_th:
+            continue
+
+        center = np.mean(cluster, axis=1)
+        if n_reg > 0:
+            # make sure the region is "far enough" from already clustered regions
+            d = np.sqrt(((np.array(list(zip(column, row, scale)))
+                            - np.tile(center.T, (n_reg, 1))) ** 2).sum(axis=1))
+            if (center[2] >= 0.6*d).sum() == 0:
+                n_reg = n_reg + 1
+                column = np.append(column, center[0])
+                row = np.append(row, center[1])
+                scale = np.append(scale, center[2])
+                gamma = np.append(gamma, t_gamma[index])
+        else:
+            n_reg = n_reg + 1
+            column = np.append(column, center[0])
+            row = np.append(row, center[1])
+            scale = np.append(scale, center[2])
+            gamma = np.append(gamma, t_gamma[index])
+
+    return np.array([row, column, scale])
+
+
+def _saliency_param(image, scales):
+    """Calculate parameters for Saliency metric.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input grayscale image
+    scales : ndarray
+        Scales for key regions
+
+    Returns
+    -------
+    entropy : (M, N) ndarray of float
+        Matrix containing entropy value of region over scales.
+    weights : (M, N) ndarray of float
+        Matrix containing abs difference of PDF over scales.
+    r : (N,) ndarray of float
+        Row index of regions.
+    c : (N,) ndarray of float
+        Column index of regions.
+    """
+    
+    image = img_as_ubyte(image)
+    nr, nc = image.shape
+    # find pixels that we are going to examine
+    mask = np.ones((nr - max(scales), nc - max(scales)))
+    r, c = np.nonzero(mask) + max(scales) / 2 + 1
+
+    n_pix = len(r)
+
+    n_scales = len(scales)
+    intensity_edges = np.arange(0, 257, 16)
+
+    previous_h = np.zeros((len(intensity_edges) - 1, n_pix))
+    weights = np.zeros((n_scales, n_pix))
+    entropy = np.zeros((n_scales, n_pix))
+
+    # iterate through every possible region
+    # iterate through scales
+    for s_idx, scale in enumerate(scales):
+        # shape for windows according to current scale
+        radius = scale + 1
+        window_shape = (radius, radius)
+
+        init = int(0 + max(scales) / 2 - scale / 2)
+        end = int(nr - max(scales) / 2 + scale / 2)
+        patches = view_as_windows(image[init:end,init:end], window_shape)
```

Comment:
Acc. to algorithm, a pixel x is taken into consideration and then the descriptor value around that pixel x with region R(defined by scale) is calculated.
For this, every pixel considered needs to have a region R for every scale, and this would be possible only till borders [max(scales)/2], so for smaller scales i take the image border from the _init_ to _end_ so that it's pixel center coincides with every scale's pixel center.